packages/wallets/particle/src/adapter.ts

```diff
@@ -74,17 +117,29 @@ export class ParticleAdapter extends BaseMessageSignerWalletAdapter {
                 throw new WalletLoadError(error?.message, error);
             }
 
-            let wallet: SolanaWallet;
+            let particleNetwork: ParticleNetwork;
+
+            const authOptions: any = {};
+            if (this._config.preferredAuthType) {
+                authOptions.preferredAuthType = this._config.preferredAuthType;
+            }
+
             try {
-                wallet = new WalletClass(new ParticleClass(this._config?.config).auth);
+                particleNetwork = new ParticleClass(this._config?.config as Config);
+                if (!particleNetwork.auth.isLogin()) {
+                    await particleNetwork.auth.login(authOptions);
+                }
             } catch (error: any) {
                 throw new WalletConfigError(error?.message, error);
             }
 
+            this._particleNetwork = particleNetwork;
```

Comment:
This should be done at the end, where we set `this._wallet`. This prevents any state changes from occurring if the initialization of anything fails in the `connect` function.

Reply:
Sure np, moved it down to where this._wallet is set.